rfc/rfc-1-q3-q4-2023-planning.md

```diff
@@ -0,0 +1,90 @@
+# RFC-1: q3-q4-2023 Planning
+
+# I Executive Summary 
+
+## A. Goal of the System
+
+Creating an open video conferencing ecosystem — i.e. open source definitions and building blocks that would support video conferencing that could be embedded in various apps — as well of course as the video conferencing system itself.
+
+For possible applications, imagine things like telemedicine apps or face-to-face customer service portals, where the video chat would be integrated rather than requiring a participant to say the equivalent of “I’ll send you a zoom link”.
+
+The Rustlemania website is a sample implementation of this system.
+
+## B. Purpose of the Request for Comment (RFC) 
+
+Define our roadmap for Q3 and Q4 of 2023
+
+## C. Overview of the Organization 
+
+Griffin Obeid: griffobeid@securityunion.dev Co-founder and developer
+
+Ronen Barzel: ronen@barzel.org Core Contributor
+
+Dario Lencina: dario@securityunion.dev  Co-founder and developer
+
+# II. Current System 
+
+## A. Description of the existing system
+
+`videocall-rs` aims to provide a world-class video conferencing system. The video conferencing software should always be open-source and MIT licensed. Also a key feature of this system is that all calls will be end-to-end encrypted between the peers in the call. 
+
+The current videocall-rs exists at [https://github.com/security-union/videocall-rs](https://github.com/security-union/videocall-rs) and is hosted at [https://rustlemania.com](https://rustlemania.com). The primary connection mechanism is using WebSockets, but the system does also support WebTransport. Everything is written in Rust with the UI being a WASM based web application built through the Yew Web Framework. The UI is very barebones at this point. The backend is highly scalable by using a pub/sub architecture with NATS we can run as many instances of the server as we want. 
+
+## B. Strengths and limitations of the current system
+
+**Strengths**
+- Highly-scalable architecture
+- Written in Rust
+- Using WebCodecs API
+- Multiple connection types supported
+- Open source & MIT licensed
+
+**Limitations**
+- Only works on Chrome or Chromium based browsers
+
+# III. Proposed System Roadmap 
+
+## A. Roadmap
+
+| Quarter | Features | Freeze Date | Release Date | Release |
+| ------- | -------- | ----------- | ------------ | ------- |
+| Q3 2023 | E2EE | 2023-08-02 | 2023-08-05 | 2 - <pick a band> |
```

Comment:
I was thinking we could end up being stuck deciding on a name for a band to use for each release. It might be easier to pull from a list like https://en.wikipedia.org/wiki/List_of_national_parks_of_the_United_States or something else, just looking to avoid decision paralysis here

Reply:
I can pick many bands I just want to give you a chance to pitch your own. How about we start with PearlJam?

Reply:
Sounds good!